lib/rubygems/command.rb

```diff
@@ -365,6 +366,28 @@ def remove_option(name)
     end
   end
 
+  def deprecate_option(short_name: nil, long_name: nil, version: nil)
+    @deprecated_options[command].merge!({ short_name => { "rg_version_to_expire" => version } }) if short_name
+    @deprecated_options[command].merge!({ long_name  => { "rg_version_to_expire" => version } }) if long_name
+  end
+
+  def check_deprecated_options(options)
+    options.each do |option|
+      if option_is_deprecated?(option)
+        version_to_expire = @deprecated_options[command][option]["rg_version_to_expire"]
+        #deprecate_option_msg = "The \"#{option}\" option has been deprecated and will be removed in future versions of Rubygems #{version_to_expire}, its use is discouraged."
```

Comment:
we shouldn't be shipping commented-out code

Reply:
Agree, sorry about that! will fix